src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,157 @@
+                            argumentList: null,
+                            initializer: SyntaxFactory.EqualsValueClause((ExpressionSyntax)expression)))));
+
+        protected override SyntaxNode GenerateLocalDeclarationStatement(string identifierTokenName, ITypeSymbol type)
+            => SyntaxFactory.LocalDeclarationStatement(
+                SyntaxFactory.VariableDeclaration(
+                    type.GenerateTypeSyntax(),
+                    SyntaxFactory.SingletonSeparatedList(SyntaxFactory.VariableDeclarator(identifierTokenName))));
+
+        protected override SyntaxNode GenerateIdentifierNameSyntaxNode(string name)
+            => SyntaxFactory.IdentifierName(name);
+
+        protected override SyntaxNode? GetInlineStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            SyntaxNode? inlineSyntaxNode = null;
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
+            {
+                var blockSyntaxNode = declarationSyntax.Body;
+                // 1. If it is a ordinary method with block
+                if (blockSyntaxNode != null)
+                {
+                    var blockStatements = blockSyntaxNode.Statements;
+                    if (blockStatements.Count == 1)
+                    {
+                        inlineSyntaxNode = GetExpressionFromStatementSyntaxNode(blockStatements[0]);
+                    }
+                }
+                else
+                {
+                    // 2. If it is using Arrow Expression
+                    var arrowExpressionNodes = declarationSyntax
+                        .DescendantNodes().Where(node => node.IsKind(SyntaxKind.ArrowExpressionClause)).ToImmutableArray();
+                    if (arrowExpressionNodes.Length == 1)
+                    {
+                        inlineSyntaxNode = ((ArrowExpressionClauseSyntax)arrowExpressionNodes[0]).Expression;
+                    }
+                }
+            }
+
+            return inlineSyntaxNode;
+        }
+
+        private static SyntaxNode? GetExpressionFromStatementSyntaxNode(StatementSyntax statementSyntax)
+            => statementSyntax switch
+            {
+                ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                _ => null
+            };
+
+        protected override bool IsEmbeddedStatementOwner(SyntaxNode syntaxNode)
+            => syntaxNode.IsEmbeddedStatementOwner();
+
+        protected override SyntaxNode GenerateTypeSyntax(ITypeSymbol symbol)
+            => symbol.GenerateTypeSyntax(allowVar: false);
+
+        protected override string GetSingleVariableNameFromDeclarationExpression(SyntaxNode syntaxNode)
+            => ((SingleVariableDesignationSyntax)((DeclarationExpressionSyntax)syntaxNode).Designation).Identifier.Text;
```

Comment:
how do you know its' a SingleVariableDesig?

Reply:
This is used to get the variable name from
`void Caller()
{
     Callee(out var name);
}`
So I am assuming it will always be a single variable. 
Would this not be true if there is syntax error?

Reply:
there's no guarantee.  and you might imagine the compiler adding `out var (x, y)` in the future.

Reply:
I see, then I feel it will be much safer to get the name from the semantic model given it is an variable declaration